### ISSUE TITLE
Wrong candidate tokens and wrong corresponding logits and scores

### ISSUE THREAD

**Issue by ljb121002** [openreasoner/openr#26]
### System Info

Standard

### Who can help?

@morning9393 @Gebro13 

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the codebase (such as scrips/, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Run following the official scripts.

### Expected behavior

Only the 'bad token' is retained, leading to the subsequent score always being 1 after applying softmax.

**Comment by xiaoweiweixiao** [comment]
meet the same error
